src/Session.cpp

```diff
@@ -8,6 +8,120 @@ Session::Session(const std::string &configFilePath):
     activeUser(nullptr)
 {
     this->fillContentFromJson(configFilePath);
+    User* defaultUser = new LengthRecommenderUser("default");
+    userMap["default"] = defaultUser;
+    activeUser = defaultUser;
+}
+
+Session::~Session()
+{
+    this->clean();
+}
+void Session::split(std::string &actionString, std::vector <std::string>& actionParams, std::string delimiter) {
+    size_t start = 0U;
+    size_t end = actionString.find(delimiter);
+    while (end != std::string::npos)
+    {
+        actionParams.push_back(actionString.substr(start, end - start));
+        start = end + delimiter.length();
+        end = actionString.find(delimiter, start);
+    }
+
+}
+
+ActionStringsEnum Session::strToEnum(const std::string &actionString) {
+    if (actionString == "createuser")
+        return createuser;
+    else if (actionString == "changeuser")
+        return changeuser;
+    else if (actionString == "deleteuser")
+        return deleteuser;
+    else if (actionString == "dupuser")
+        return dupuser;
+    else if (actionString == "content")
+        return listContent;
+    else if (actionString == "watchlist")
+        return watchlist;
+    else if (actionString == "watch")
+        return watch;
+    else if (actionString == "log")
+        return logActions;
+    else
+        return exitLoop; //if actionString == "exit" or invalid input, program will exit
+}
+
+void Session::start() {
+    std::string actionString;
+    std::vector<std::string> actionParams;
+    ActionStringsEnum action;
+    do {
+        std::getline(std::cin, actionString);
+        actionParams.clear();
+        split(actionString, actionParams, ACTION_PARAMS_DELIMITER);
+        action = strToEnum(actionParams[0]);
+        BaseAction* actionObj;
+        switch (action) {
+            case createuser: {
+                actionObj = new CreateUser(actionParams[1], actionParams[2]); //TODO do we assume valid input?
+                actionsLog.push_back(actionObj);
+                actionObj->act(*this);
+            }
```

Comment:
First, since push_back and act are run for all of them, maybe you can find a better way of doing this (with less repeating code).
Second, add a `break;` or else the switch will leak

Reply:
why would it leak?.. we know only one case happens each time.
It's not the same object Creation, and Watch is not only push and act so its problematic
and this way is better cause its easy to change in future

Reply:
You're right, it wouldn't. My mistake